hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -243,20 +239,24 @@ object LoweredTableReader {
       "globals",
       scanBody(Ref("context", contextType)))
 
-    val partDataWithIndex = InsertFields(
-      ArrayRef(Ref("scanResult", scanResult.typ), Ref("i", TInt32)),
-      FastIndexedSeq(
-        "i" -> Ref("i", TInt32)))
-    val sortedPartDataIR = ArraySort(
-      Let("scanResult", scanResult,
-        StreamMap(
-          StreamRange(I32(0), ArrayLen(Ref("scanResult", scanResult.typ)), I32(1)),
-          "i",
-          partDataWithIndex)),
-      "l", "r",
-      ApplyComparisonOp(LT(keyType),
-        GetField(Ref("l", partDataWithIndex.typ), "minkey"),
-        GetField(Ref("r", partDataWithIndex.typ), "minkey")))
+    val sortedPartDataIR = sortIR(bindIR(scanResult) { scanResult =>
+      mapIR(
+        filterIR(
+          mapIR(
+            rangeIR(I32(0), ArrayLen(scanResult))) { i =>
+            InsertFields(
+              ArrayRef(scanResult, i),
+              FastIndexedSeq("i" -> i))
+          }) { row => ArrayLen(GetField(row, "minkey")) > 0 }
+      ) { row =>
+        InsertFields(row, FastSeq(
+          ("minkey", ArrayRef(GetField(row, "minkey"), I32(0))),
+          ("maxkey", ArrayRef(GetField(row, "maxkey"), I32(0)))))
+      }
+    }) { (l, r) =>
+      ApplyComparisonOp(LT(keyType), GetField(l, "minkey"), GetField(r, "minkey"))
```

Comment:
You didn't introduce this, but shouldn't this sort on `(key, i)` lexicographically? Say the partitions are `[0, 2], [2, 2], [2, 5]`, and it sorted by min key to `[0, 2], [2, 5], [2, 2]`. Then we would decide the data was unsorted.

Reply:
I can fix this. I think it should sort on `(minkey, maxkey)` though -- the index shouldn't matter.

I believe the adversarial case is:
```
[0, 2], [2, 5], [2, 2]
```
This will be sorted to itself, and will fail the sortedness check in the next phase.

Reply:
Yup, you're right.

Reply:
done